ql/src/java/org/apache/hadoop/hive/ql/ddl/ShowUtils.java

```diff
@@ -0,0 +1,397 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.ql.ddl;
+
+import com.google.common.collect.Lists;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.common.type.HiveDecimal;
+import org.apache.hadoop.hive.common.type.Timestamp;
+import org.apache.hadoop.hive.metastore.api.BinaryColumnStatsData;
+import org.apache.hadoop.hive.metastore.api.BooleanColumnStatsData;
+import org.apache.hadoop.hive.metastore.api.ColumnStatisticsData;
+import org.apache.hadoop.hive.metastore.api.ColumnStatisticsObj;
+import org.apache.hadoop.hive.metastore.api.DateColumnStatsData;
+import org.apache.hadoop.hive.metastore.api.Decimal;
+import org.apache.hadoop.hive.metastore.api.DecimalColumnStatsData;
+import org.apache.hadoop.hive.metastore.api.DoubleColumnStatsData;
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.metastore.api.LongColumnStatsData;
+import org.apache.hadoop.hive.metastore.api.StringColumnStatsData;
+import org.apache.hadoop.hive.metastore.api.TimestampColumnStatsData;
+import org.apache.hadoop.hive.ql.exec.Utilities;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.serde2.io.DateWritableV2;
+import org.apache.hadoop.hive.serde2.io.TimestampWritableV2;
+import org.apache.hive.common.util.HiveStringUtils;
+import org.codehaus.jackson.map.ObjectMapper;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.math.BigInteger;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+/**
+ * Utilities for SHOW ... commands.
+ */
+public final class ShowUtils {
+  private ShowUtils() {
+    throw new UnsupportedOperationException("ShowUtils should not be instantiated");
+  }
+
+  public static final Charset UTF_8 = Charset.forName("UTF-8");
+
+  public static DataOutputStream getOutputStream(Path outputFile, DDLOperationContext context) throws HiveException {
+    try {
+      FileSystem fs = outputFile.getFileSystem(context.getConf());
+      return fs.create(outputFile);
+    } catch (Exception e) {
+      throw new HiveException(e);
+    }
+  }
+
+  public static String propertiesToString(Map<String, String> props, Set<String> exclude) {
+    if (props.isEmpty()) {
+      return "";
+    }
+  
+    SortedMap<String, String> sortedProperties = new TreeMap<String, String>(props);
+    List<String> realProps = new ArrayList<String>();
+    for (Map.Entry<String, String> e : sortedProperties.entrySet()) {
+      if (e.getValue() != null && (exclude == null || !exclude.contains(e.getKey()))) {
```

Comment:
I personally hate NULL values.  Can you get rid of this check (`exclude == null`) and simply call this method with `Collections.emptySet()` ?

Reply:
Totally agree, fixed.